### PR TITLE
make strict the default option for choice validation

### LINKED ISSUE #7171: Add docs for making strict the default option for choice validation
In https://github.com/symfony/symfony/pull/19257 we made some changes for the Choice validation:

```
Setting the strict option of the `Choice` Constraint to `false` has been  
deprecated and the option will be changed to `true` as of 4.0.
```

### PR DESCRIPTION
**Pull request by snoek09**
This fixes https://github.com/symfony/symfony-docs/issues/7171.

### CODE REVIEW

File: reference/constraints/Choice.rst [hunk 1]
```diff
@@ -367,10 +367,13 @@ too many options per the `max`_ option.
 strict
 ~~~~~~
 
-**type**: ``boolean`` **default**: ``false``
+**type**: ``boolean`` **default**: ``true``
+
+The validator will also check the type of the input value.
```
Comment (wouterj) [inline reference/constraints/Choice.rst]: Wouldn't it make more sense to keep the current description? Or at least tell a bit about what "check the type" means?

File: reference/constraints/Choice.rst [hunk 2]
```diff
@@ -367,10 +367,13 @@ too many options per the `max`_ option.
...
+
+.. caution::
 
-If true, the validator will also check the type of the input value. Specifically,
-this value is passed to as the third argument to the PHP :phpfunction:`in_array`
-method when checking to see if a value is in the valid choices array.
+    Setting the strict option of the `Choice` Constraint to `false` has been
+    deprecated as of Symfony 3.2 and the option will be changed to `true` as of 4.0.
```
Comment (wouterj) [inline reference/constraints/Choice.rst]: As it's reStructed Text, you have to use double backticks for literals: ``` ``true`` ```

Also, I suggest removing the literal from "Choice constraint". Literals kinda break the read flow, which makes things a little harder to read.

**Review by javiereguiluz — APPROVED** [review]
👍

**Comment by xabbuh** [comment]
Thank you @snoek09. I have merged your PR into the `3.2`. Afterwards, I realised that we shouldn't have changed the default value as it's just deprecated right now to set it to `false`, but the default value will stay untouched until Symfony 4.0. So I addressed this in 80eaad355c824277cf6f4284a9191fab805eec4e.

**Pull request closed without merging**